<launch>

  <arg name="tf_prefix"         default="mur620a" />
  <arg name="UR_prefix"  default="UR10_l" />
  <arg name="relative_pose"          default="[0.5, 0.0, 0.0, 0.0, 0.0, 0.0]" />
  <!-- <arg name="admittance" default="[0.01,0.01,0.001,0.1,0.1,0.1]" /> -->
  <arg name="admittance" default="[0.001,0.001,0.001,0.0,0.0,0.0]" />
  
  <arg name="object_pose_topic" default="/virtual_object/object_pose" />
  <arg name="object_vel_topic" default="/virtual_object/object_vel" />
  <arg name="manipulator_global_pose_topic" default="/$(arg tf_prefix)/$(arg UR_prefix)/global_tcp_pose" />
  <arg name="manipulator_local_pose_topic" default="/$(arg tf_prefix)/$(arg UR_prefix)/ur_calibrated_pose" />
  <arg name="manipulator_vel_topic" default="manipulator_vel" />
  <arg name="manipulator_command_topic" default="/$(arg tf_prefix)/$(arg UR_prefix)/twist_controller/command" />
  <arg name="wrench_topic" default="/$(arg tf_prefix)/$(arg UR_prefix)/wrench" />
  <arg name="mir_pose_topic" default="/$(arg tf_prefix)/mir_pose_simple" />
  <arg name="mir_cmd_vel_topic" default="/$(arg tf_prefix)/cmd_vel" />
  <arg name="manipulator_base_frame" default="$(arg tf_prefix)/$(arg UR_prefix)/base_link" />
  <arg name="mir_base_frame" default="$(arg tf_prefix)/base_link" />
  <arg name="relative_pose_topic" default="/$(arg tf_prefix)/$(arg UR_prefix)/relative_pose" />



  <group ns="$(arg tf_prefix)/$(arg UR_prefix)">
    <!-- <rosparam file="$(find virtual_object)/config/virtual_object.yaml" command="load" /> -->

    <node name="dezentralized_admittance_controller" pkg="manipulator_control" type="dezentralized_admittance_controller.py" output="screen">
      <rosparam param="relative_pose" subst_value="True">$(arg relative_pose) </rosparam>
      <rosparam param="admittance" subst_value="True">$(arg admittance) </rosparam>
      <param name="tf_prefix" type="string" value="$(arg tf_prefix)"/>
      <param name="object_pose_topic" type="string" value="$(arg object_pose_topic)"/>
      <param name="object_vel_topic" type="string" value="$(arg object_vel_topic)"/>
      <param name="manipulator_global_pose_topic" type="string" value="$(arg manipulator_global_pose_topic)"/>
      <param name="manipulator_local_pose_topic" type="string" value="$(arg manipulator_local_pose_topic)"/>
      <param name="manipulator_vel_topic" type="string" value="$(arg manipulator_vel_topic)"/>
      <param name="manipulator_command_topic" type="string" value="$(arg manipulator_command_topic)"/>
      <param name="wrench_topic" type="string" value="$(arg wrench_topic)"/>
      <param name="mir_pose_topic" type="string" value="$(arg mir_pose_topic)"/>
      <param name="mir_cmd_vel_topic" type="string" value="$(arg mir_cmd_vel_topic)"/>
      <param name="manipulator_base_frame" type="string" value="$(arg manipulator_base_frame)"/>
      <param name="ur_prefix" type="string" value="$(arg UR_prefix)"/>
      <param name="mir_base_frame" type="string" value="$(arg mir_base_frame)"/>
      <param name="relative_pose_topic" type="string" value="$(arg relative_pose_topic)"/>
    </node>
  </group>

</launch>
